source/extensions/omni.isaac.lab/omni/isaac/lab/sensors/frame_transformer/frame_transformer.py

```diff
@@ -206,37 +205,75 @@ def _initialize_impl(self):
+
         # Create simulation view
         self._physics_sim_view = physx.create_simulation_view(self._backend)
         self._physics_sim_view.set_subspace_roots("/")
         # Create a prim view for all frames and initialize it
         # order of transforms coming out of view will be source frame followed by target frame(s)
-        self._frame_physx_view = self._physics_sim_view.create_rigid_body_view(body_names_regex.replace(".*", "*"))
+        self._frame_physx_view = self._physics_sim_view.create_rigid_body_view(body_names_regex)
 
         # Determine the order in which regex evaluated body names so we can later index into frame transforms
         # by frame name correctly
         all_prim_paths = self._frame_physx_view.prim_paths
 
-        # Only need first env as the names and their ordering are the same across environments
-        first_env_prim_paths = all_prim_paths[0 : len(tracked_body_names)]
-        first_env_body_names = [first_env_prim_path.split("/")[-1] for first_env_prim_path in first_env_prim_paths]
+        if "env_" in all_prim_paths[0]:
+
+            def extract_env_num_and_prim_path(item: str) -> tuple[int, str]:
+                """Separates the environment number and prim_path from the item.
+
+                Args:
+                    item: The item to extract the environment number from. Assumes item is of the form
+                        `/World/env_1/blah` or `/World/env_11/blah`.
+                Returns:
+                    The environment number and the prim_path.
+                """
+                match = re.search(r"env_(\d+)(.*)", item)
+                return (int(match.group(1)), match.group(2))
+
+            # Find the indices that would reorganize output to be per environment. We want `env_1/blah` to come before `env_11/blah`
+            # and env_1/Robot/base to come before env_1/Robot/foot so we need to use custom key function
+            self._per_env_indices = [
+                index
+                for index, _ in sorted(
+                    list(enumerate(all_prim_paths)), key=lambda x: extract_env_num_and_prim_path(x[1])
+                )
+            ]
+
+            # Only need 0th env as the names and their ordering are the same across environments
+            sorted_prim_paths = [
+                all_prim_paths[index] for index in self._per_env_indices if "env_0" in all_prim_paths[index]
+            ]
+
+        else:
+            # If no environment is present, then the order of the body names is the same as the order of the prim paths sorted alphabetically
+            self._per_env_indices = [index for index, _ in sorted(enumerate(all_prim_paths), key=lambda x: x[1])]
+            sorted_prim_paths = [all_prim_paths[index] for index in self._per_env_indices]
+
+        # -- target frames
+        self._target_frame_body_names = [prim_path.split("/")[-1] for prim_path in sorted_prim_paths]
 
```

Comment:
I needed to add this in to take care of the case where we don't have `env_` (for example our standalone `run_frame_transformer.py` script)